integration_tests/can_launch/test_driver/main_test.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+main() {
+  group('can launch an app', () {
+    FlutterDriver driver;
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+    });
+
+    tearDownAll(() async {
+      if (driver != null) driver.close();
+    });
+
+    // Simple check that the app is responding to UI actions.
+    test('tap on the floating action button; verify counter', () async {
+      // Find floating action button (fab) to tap on
+      ObjectRef fab = await driver.findByTooltipText('Increment');
```

Comment:
That's a good point. @abarth and I chatted about this. The consensus was that we don't want to set keys on widgets _only_ for testing purposes, and decided to look by text instead.

What if we add a `debugKey` to widget that could be tree-shaken off in production? I've used the following trick in the past for code slated for compilation to JS:

```dart
// the number of AST nodes in this function is low enough for
// dart2js to inline it.
bool assertionsEnabled() {
  var k = false;  // has to be `var`, cause `bool` adds an extra AST node
  assert((k = true));
  return k;
}

if (assertionsEnabled()) {
  // code here will be tree-shaken off because the
  // `if` condition evals to `false` in prod mode at compile time.
}
```

Similarly we could do:

```dart
// evals to `null` in prod mode at compile time
dynamic createDebugKey(dynamic key) {
  return assertionsEnabled() ? key : null;
}

build(_) {
  return new Text(
    'some text',
    debugKey: createDebugKey(123)
  );
}
```